libs/community/langchain_community/vectorstores/upstash.py

```diff
@@ -0,0 +1,629 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import TYPE_CHECKING, Iterable, List, Optional, Tuple
+
+import numpy as np
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.utils.iter import batch_iterate
+from langchain_core.vectorstores import VectorStore
+
+from langchain_community.vectorstores.utils import (
+    maximal_marginal_relevance,
+)
+
+if TYPE_CHECKING:
+    from upstash_vector import AsyncIndex, Index
+
+logger = logging.getLogger(__name__)
+
+
+class UpstashVectorStore(VectorStore):
+    """Upstash Vector vector store.
+
+    To use, you should have the ``upstash_vector`` python package installed.
+
+    Create a new Upstash Vector database and copy the `index_url` and `index_token`.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.vectorstores.upstash import UpstashVectorStore
+            from langchain_community.embeddings.openai import OpenAIEmbeddings
```

Comment:
can simplify
```python
from langchain_community.vectorstores import...
from langchain_community.embeddings import...
```

Reply:
When using folder exports the types turn into `Any`. I left it like that since I thought it would be more helpfull for people copy pasting it. But I can remove it if thats ok

<img width="509" alt="image" src="https://github.com/langchain-ai/langchain/assets/18082091/7aa2e970-a5f4-4169-85dc-b3a1778d6ab4">

<img width="549" alt="image" src="https://github.com/langchain-ai/langchain/assets/18082091/218a4e8d-9786-47aa-b167-d41d59057633">